iothub/device/src/TransportSettings/IotHubClientMqttSettings.cs

```diff
@@ -95,6 +95,16 @@ public IotHubClientMqttSettings(IotHubClientTransportProtocol transportProtocol
         /// </remarks>
         public RemoteCertificateValidationCallback RemoteCertificateValidationCallback { get; set; }
 
+        /// <summary>
+        /// Cloud-to-device message queue size for storing offline messages when the client connects with <see cref="CleanSession" /> set to <c>false</c>.
+        /// </summary>
+        /// <remarks>
+        /// If incoming message callback is not set, messages from service are received and stored in a queue until a callback is set. 
+        /// If the number of messages sent is greater than queue size, older messages are removed as newer messages are added. 
+        /// Setting this to a low value can cause offline undelivered messages to get lost.
+        /// </remarks>
+        public int IncomingMessageQueueSize { get; set; } = 10;
```

Comment:
Can you add a reference to this property to the public `SetIncomingMessageCallbackAsync` as well. We want people to know about this property when the set the callback (even if they don't want to change the default value).